### PR TITLE
Add monadic utility functions

### PR DESCRIPTION
**Pull request by giacomocavalieri**
This PR adds some monadic utility functions to improve the readability of code written using MTL-style effects.

### CODE REVIEW

**Comment by giacomocavalieri** [comment]
As a side note: writing the code I notices wart remover getting really angry with the monadic `unless` and `when` because it kept inferring the type `Any`; we have to keep an eye on this I think we'll have to disable this Wart to keep the code readable: 
```scala
when (updatedOrder needsMoreOf product) (emit(MissingProduct(product)))
// would turn into
when[<type annotation>] (updatedOrder needsMoreOf product) (emit(MissingProduct(product)))
// The type annotation is quite ugly and breaks the flow of the sentence when reading the code
```

File: utils/src/main/scala/dev/atedeg/mdm/utils/Monads.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package dev.atedeg.mdm.utils
+
+import cats.data.NonEmptyList
+import cats.mtl.{Raise, Tell}
```
Comment (nicolasfara) [inline utils/src/main/scala/dev/atedeg/mdm/utils/Monads.scala:4]: Scalafmt or scalafix (I don't remember which one) should add a space after and before the curly braces.
It's strange that the CI can't catch it.
Could you try to do a `scalafmtAll` and `scalafixAll` from sbt and see if the space is added?

**Review by nicolasfara — APPROVED** [review]

File: utils/src/main/scala/dev/atedeg/mdm/utils/Monads.scala (hunk 1 above)
Comment (giacomocavalieri) [reply utils/src/main/scala/dev/atedeg/mdm/utils/Monads.scala:4]: I tried running both but they did not reformat the import list

**Review by ndido98 — APPROVED** [review]

File: utils/src/main/scala/dev/atedeg/mdm/utils/Monads.scala (hunk 1 above)
Comment (nicolasfara) [reply utils/src/main/scala/dev/atedeg/mdm/utils/Monads.scala:4]: Ok, no problem. With #96 we fix it.

**Review by vitlinda — APPROVED** [review]

**Pull request merged**